Apeswap-Bots/New-Pair-Creation/src/agent.ts

```diff
@@ -0,0 +1,45 @@
+import { Finding, HandleTransaction, TransactionEvent, getEthersProvider } from "forta-agent";
+import { providers } from "ethers";
+
+import { APEFACTORY_ABI } from "./constants";
+import NetworkManager from "./network";
+import NetworkData from "./network";
+
+import { createFinding, newPairParamsType, newPairFindingType, providerParams } from "./utils";
+const { CREATE_PAIR_FUNCTION } = APEFACTORY_ABI;
+
+const networkManager = new NetworkManager();
+
+export const initialize = (provider: providers.Provider) => {
+  return async () => {
+    const { chainId } = await provider.getNetwork();
+    networkManager.setNetwork(chainId);
+  };
+};
+
+export const provideHandleTransaction = (
+  { functionSig }: newPairParamsType,
+  { apeFactoryAddress }: NetworkData
```

Comment:
this deconstruction is de-referencing the factoryAddress so when the internal NetworkData address, the value is not going to be reflected in the bot. So as the default value is `""` then the address will be `""` always because the change made during initialization is not going to be propagated